src/basicfuns.jl

```diff
@@ -190,14 +190,21 @@ function logsumexp(X)
     isempty(X) && return log(sum(X))
     reduce(logaddexp, X)
 end
-function logsumexp(X::AbstractArray{T}) where {T<:Real}
+function logsumexp(X::AbstractVector{T}) where {T<:Real}
```

Comment:
This function is now redundant with the new one.

Reply:
I don't have the corresponding line for `isfinite(u) || return float(u)` in the vector one. Any idea how to implement it?

Reply:
Maybe `u isa AbstractArray || isfinite(u) || return float(u)`?

Reply:
Yes, sounds good. Basically, this is just an optimization to avoid computing the value when we already know it will be `NaN` or `Inf`, right?

Reply:
I believe so. I will unify the two by doing what I proposed.

Reply:
Done.